src/Shared/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -87,6 +81,104 @@ public async Task<IDataReader> ExecuteKustoQueryAsync(KustoQuery query)
             }
         }
 
+        /// <summary>
+        /// Use this method to receive large quantities of data from Kusto in a "stream". 
+        /// This method assumes the query is returning a single schema result set. 
+        /// </summary>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public async IAsyncEnumerable<object[]> ExecuteStreamableKustoQuery(KustoQuery query)
+        {
+            var client = GetProvider();
+            var properties = BuildClientRequestProperties(query);
+            properties.SetOption(ClientRequestProperties.OptionResultsProgressiveEnabled, true);
+
+            string text = BuildQueryText(query);
+
+            ProgressiveDataSet pDataSet = await client.ExecuteQueryV2Async(
+                DatabaseName,
+                text,
+                properties);
+
+            int tableCompletionCount = 0;
+
+            using IEnumerator<ProgressiveDataSetFrame> frames = pDataSet.GetFrames();
```

Comment:
Hm, how does this work? `IEnumerator` doesn't implement `IDisposable`. Does `GetFrames()` return something that does?

Reply:
I mean, it works without it... Maybe @ChadNedzlek knows something we don't :)

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.ienumerator-1?view=net-6.0 disagrees?

Reply:
Ah, it's the non-generic one that doesn't (which isn't the one we are using)

Reply:
Oh! Sorry, I was sloppy in my reading. Carry on.

Reply:
I had exactly the same spit take.  Google failed me!